### DISCUSSION TITLE
balancing a dataset

### DISCUSSION THREAD

**Discussion by Jameel9** [chemprop/chemprop#1106] [General]
Hi everyone,

Q1. Does Chemprop really need the input dataset to be balanced for a binary classification task?
I have a dataset in which one of the classes is 3.14% of the total dataset. I tried to eliminate every ith row from the majority class in order to balance the dataset. However, that created a clear difference in the distribution of the data points (after balancing) between classes (as seen in the image). Generating synthetic data (oversampling) is not an option because I am not convinced by its accuracy. 

![Pasted image](https://github.com/user-attachments/assets/1c6b5e5a-076e-415d-8699-f01949fd53a7)

So, Q2 Does that difference in distribution affect the model accuracy? If yes, what could be done to undersample the majority class and retain a smooth distribution

Thank you for your help!

Jameel

**Comment by JacksonBurns** [comment]
Needing balanced data isn't really a Chemprop-specific problem. Broadly speaking deep learning models require balanced training data because they tend to overfit to the majority class otherwise. We often say that the output of the network being above 0.5 is positive and below is negative, so you can imagine that if 97% of the data is of only one class this threshold starts to make less sense. In that sense, the 'distribution' of the training data has a significant impact.

I'm not quite sure that you are showing in this plot and what you mean by 'distribution' in this context, though. Your training data should be a set of exclusively 1/0 (in the case of binary classification). If you are binning a continuous variable to cast the problem as classification, consider adjusting your bins to achieve a balance.

There are also other more advanced methods to deal with class imbalance besides what you have tried - check out: https://imbalanced-learn.org/stable/index.html
